spring-integration-sftp/src/test/java/org/springframework/integration/sftp/session/SftpServerTests.java

```diff
@@ -125,7 +127,12 @@ private void testKeyExchange(String pubKey, String privKey, String passphrase)
 
 	private PublicKey decodePublicKey(String key) throws Exception {
 		InputStream stream = new ClassPathResource(key).getInputStream();
-		byte[] decodeBuffer = Base64.decodeBase64(StreamUtils.copyToByteArray(stream));
+		byte[] keyBytes = StreamUtils.copyToByteArray(stream);
+		// strip any newline chars
+		while (keyBytes[keyBytes.length - 1] == 0x0a || keyBytes[keyBytes.length - 1] == 0x0c) {
```

Comment:
See my comment on the JIRA - problem is unrelated but we should do the upgrade anyway.